cpp/include/cudf/utilities/traits.hpp

```diff
@@ -69,6 +81,24 @@ constexpr inline bool is_relationally_comparable()
   return is_relationally_comparable_impl<L, R>::value;
 }
 
+/**
+ * @brief Indicates whether objects of types `L` and `R` can be compared
+ * for equality.
+ *
+ * Given two objects `L l`, and `R r`, returns true if `l == r` is a
+ * well-formed expression.
+ *
+ * @tparam L Type of the first object
+ * @tparam R Type of the second object
+ * @return true Objects of types `L` and `R` can be compared for equality
+ * @return false Objects of types `L` and `R` cannot be compared
+ */
+template <typename L, typename R>
+constexpr inline bool is_equality_comparable()
```

Comment:
The "is" in the name is giving my internal grammarian fits, since there are two things to compare. :)

Reply:
True.  I point to "prior art" with is_relationally_comparable  :)